lib/lightning/scrubber.ex

```diff
@@ -79,8 +96,10 @@ defmodule Lightning.Scrubber do
   Prepare a list of sensitive samples (strings) into a potentially bigger list
   composed of variations a sample may appear.
   """
-  @spec encode_samples(samples :: [String.t()]) :: [String.t()]
-  def encode_samples(samples) do
+  @spec encode_samples(samples :: [String.t()], basic_auth :: [String.t()]) :: [
+          String.t()
+        ]
+  def encode_samples(samples, basic_auth \\ []) do
```

Comment:
Hey @stuartc , this is needed because the `username` and `email` are on the safe list of the `SensitiveValues` in order catch `Basic` auth method on request hearders.

It was caught by this additional assertion line:

```
MapSet.new([
          {"R/T", "Starting operation 1"},
          {"JOB", "#{expected_job_x_value}"},
          # Check to ensure that an inadvertantly exposed secret from job 2 is
          # still scrubbed properly in job 3.
          {"JOB", "quux is on the safelist"},
          {"JOB", "but *** should be scrubbed"},
          {"JOB", "along with its encoded form ***"}, <<<---
          {"R/T", "Expression complete!"}
        ])
```
when a run logs something like:
```
    "fn(state => {
      state.x = state.x * 3;
      console.log(state.x);
      console.log('quux is on the safelist')
      console.log('but immasecret should be scrubbed');
      console.log('along with its encoded form #{Base.encode64("quux:immasecret")}');
      return state;
    });"
```
where `quux:immasecret` is the `<username>:<password>` (same applies to a Basic auth with email and password so we have this combinations provided by `Credentials.basic_auth_for(%Credential{body: body})`)